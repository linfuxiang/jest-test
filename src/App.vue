<template>
  <div id="app">
    <h1>{{ msg }}</h1>

    <input v-model="inputCpm1">
    <cmp1 :msg="inputCpm1"></cmp1>

    <button @click="show=!show">c'mon in</button>
    <div class="come" v-show="show">Hey, i'm coming</div>
  </div>
</template>

<script>
import cmp1 from './components/cmp1'
import axios from 'axios'
export default {
  name: 'app',
  data () {
    return {
      msg: 'Welcome to Your Vue.js App',
      inputCpm1: '',
      show: false,
    }
  },
  components: {
    cmp1,
  },
  methods: {
  },
  mounted() {
  },
}
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}

h1, h2 {
  font-weight: normal;
}

ul {
  list-style-type: none;
  padding: 0;
}

li {
  display: inline-block;
  margin: 0 10px;
}

a {
  color: #42b983;
}
</style>
