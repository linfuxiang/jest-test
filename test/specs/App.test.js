import { shallow } from 'vue-test-utils'
import Vue from 'vue'
import App from '@/App'

// describe('App.vue', () => {
// 	let cons, vm
	
// 	beforeAll(()=>{
// 		cons = Vue.extend(App)
// 		vm = new cons().$mount()
// 	})
	
// 	it('test Data', () => {
// 		expect(vm.msg).toBe('Welcome to Your Vue.js App')
// 		expect(vm.showingCmp).toBe('cmp1')
// 	})
// 	it('test checkoutCmp func', () => {
// 		vm.checkoutCmp()
// 		expect(vm.showingCmp).toBe('cmp2')
// 		vm.checkoutCmp()
// 		expect(vm.showingCmp).toBe('cmp1')
// 	})
// 	it('has the expected html structure', () => {
//   		expect(vm.$el).toMatchSnapshot()
// 	})
// })

describe('just test App.vue', ()=>{
	let cmp
	
	beforeAll(()=>{
		cmp = shallow(App, {})
	})

	it('test Data', ()=>{
		expect(cmp.vm.msg).toBe('Welcome to Your Vue.js App')
		expect(cmp.vm.showingCmp).toBe('cmp1')
	})
	it('test checkoutCmp func', () => {
		cmp.vm.checkoutCmp()
		expect(cmp.vm.showingCmp).toBe('cmp2')
		cmp.vm.checkoutCmp()
		expect(cmp.vm.showingCmp).toBe('cmp1')
	})
	it('snapshot', () => {
  		expect(cmp.element).toMatchSnapshot()
	})
})