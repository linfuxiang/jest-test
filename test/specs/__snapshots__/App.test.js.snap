// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`just test App.vue snapshot 1`] = `
<div
  id="app"
>
  <h1>
    Welcome to Your Vue.js App
  </h1>
   
  <!---->
   
  <button>
    checkout component
  </button>
</div>
`;
